File: go/2348_Number_of_Zero_Filled_Subarrays.go
func zeroFilledSubarray(nums []int) int64 {
    var totalSubarrays int64
    var currentSubarrays int64

    for _, value := range nums {
        if (value == 0) {
            currentSubarrays++
            totalSubarrays += currentSubarrays;
        } else {
            currentSubarrays = 0;
        }
    }

    return totalSubarrays
}

File: ts/2348_Number_of_Zero-Filled_Subarrays.ts
function zeroFilledSubarray(nums: number[]): number {
    let currentSubarrays: number = 0;
    let totalSubarrays: number = 0;
    nums.forEach((num) => {
        if (num == 0) {
            currentSubarrays++;
            totalSubarrays += currentSubarrays;
        } else {
            currentSubarrays = 0;
        }
    });

    return totalSubarrays;
};
